### PR TITLE
feat: enable model naming.

### PR DESCRIPTION
**Pull request by sayakpaul**
Do refer to the notebook introduced in the PR. We need to name each layer so as to initialize the student model from the teacher. 

Additionally, I am running [some benchmarks](https://github.com/sayakpaul/compressed-mirnet/wiki/Benchmarking-teacher-and-student-models-on-Pixel-6) with the TFLite models to estimate how the student model would run on mobile phones.

### CODE REVIEW

**Comment by sayakpaul** [comment]
@soumik12345 if https://github.com/sayakpaul/compressed-mirnet/issues/8 is incorporated, I am afraid we'd need to train the teacher model once again.

**Comment by soumik12345** [comment]
> @soumik12345 if #8 is incorporated, I am afraid we'd need to train the teacher model once again.

@sayakpaul Let's create a minimal training script + config system in a separate PR.

File: compressed_mirnet/model/blocks/recursive_residual.py [hunk 1]
```diff
@@ -1,11 +1,10 @@
-import tensorflow as tf
 from tensorflow.keras import layers
 
 from .attention import DualAttention
 from .skff import SelectiveKernelFeatureFusion
 
 
-def DownSampling():
+def DownSampling(idx):
```
Comment (soumik12345) [inline compressed_mirnet/model/blocks/recursive_residual.py:7]: Is `idx` supposed to be an integer index? Or is it just a prefix for layer names?

**Review by soumik12345 — APPROVED** [review]
The changes look good to me.

File: compressed_mirnet/model/blocks/recursive_residual.py (hunk 1 above)
Comment (sayakpaul) [reply compressed_mirnet/model/blocks/recursive_residual.py:7]: String. Just a name identifier.

Comment (sayakpaul) [reply compressed_mirnet/model/blocks/recursive_residual.py:7]: Am I good to merge?

**Pull request merged**